runtime/src/deterministic.rs

```diff
@@ -60,8 +53,25 @@ use tracing::trace;
 /// Range of ephemeral ports assigned to dialers.
 const EPHEMERAL_PORT_RANGE: Range<u16> = 32768..61000;
 
+/// Label for root task created during `Runner::start`.
 const ROOT_TASK: &str = "root";
 
+/// Seed to use for random number generation.
+#[derive(Clone)]
+pub enum Seed {
+    /// Initialize a new random number generator with the given seed.
+    Number(u64),
+    /// Sampler allows a previously initialized random number generator
+    /// to be used.
+    ///
+    /// This is useful when spawning multiple runtimes from the same
+    /// base seed (typically done to mock unclean exit).
+    Sampler(Arc<Mutex<StdRng>>),
```

Comment:
Open question: is it better to have a built-in "reset" that keeps the same runtime alive across simulated unclean shutdowns or is it best to drop and instantiate a new runtime each time?

Reply:
We'd want to retain the ability to pre-populate `storage` in either case.

Reply:
`export`/`fork` should be used here such that caller doesn't need to be aware of which internal variables need to be retained to keep going..